src/jit/lower.h

```diff
@@ -55,27 +55,63 @@ class Lowering : public Phase
             LowerNode(newNode);
         }
     }
+    void LowerRange(GenTree* firstNode, GenTree* lastNode)
+    {
+        for (GenTree* node = firstNode; node != lastNode->gtNext; node = node->gtNext)
+        {
```

Comment:
I would make `LIR::ReadOnlyRange::ReadOnlyRange(GenTree* firstNode, GenTree* lastNode)` public and use a `ReadOnlyRange` instead of a `Range`. The former does not offer the same guarantees as the latter, which is why it does not support the various mutators. We already use this in other cases where we need to iterate over a subrange (e.g. `NonPhiNodes`).